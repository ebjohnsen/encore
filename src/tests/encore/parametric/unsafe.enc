unsafe class Duplicator[unsafe t]
  var f : t
  def duplicate() : (t, t)
    (this.f, this.f)
  end
end

linear class Token
end

active class Main
  def main(args : [String]) : unit
    val duplicator = new Duplicator[Token]
    duplicator.f = new Token
    match duplicator.duplicate() with
      case (y, y') => println("All your linear are belong to us")
    end
  end
end