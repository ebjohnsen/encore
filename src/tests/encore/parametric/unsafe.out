All your linear are belong to us
